scripts/exglobal_atmos_sfcanl.sh

```diff
@@ -22,198 +22,127 @@
-    export MAX_TASKS_CY=${ntiles}
-
-    CDATE="${PDY}${cyc}" ${CYCLESH}
-    export err=$?; err_chk
-fi
-
-# Update surface restarts at middle of window
-for n in $(seq 1 ${ntiles}); do
-    if [[ ${DO_JEDISNOWDA:-"NO"} = "YES" ]]; then
-        ${NCP} "${COM_SNOW_ANALYSIS}/${PDY}.${cyc}0000.sfc_data.tile${n}.nc" \
-               "${COM_ATMOS_RESTART}/${PDY}.${cyc}0000.sfcanl_data.tile${n}.nc"
-    else
-        ${NCP} "${COM_ATMOS_RESTART_PREV}/${PDY}.${cyc}0000.sfc_data.tile${n}.nc" \
-               "${COM_ATMOS_RESTART}/${PDY}.${cyc}0000.sfcanl_data.tile${n}.nc"
-    fi
-    ${NLN} "${COM_ATMOS_RESTART_PREV}/${PDY}.${cyc}0000.sfc_data.tile${n}.nc" "${DATA}/fnbgsi.00${n}"
-    ${NLN} "${COM_ATMOS_RESTART}/${PDY}.${cyc}0000.sfcanl_data.tile${n}.nc"   "${DATA}/fnbgso.00${n}"
-    ${NLN} "${FIXgfs}/orog/${CASE}/${CASE}_grid.tile${n}.nc"                       "${DATA}/fngrid.00${n}"
-    ${NLN} "${FIXgfs}/orog/${CASE}/${CASE}.mx${OCNRES}_oro_data.tile${n}.nc"                   "${DATA}/fnorog.00${n}"
-done
-
+# global_cycle executable specific variables
 export APRUNCY=${APRUN_CYCLE}
 export OMP_NUM_THREADS_CY=${NTHREADS_CYCLE}
 export MAX_TASKS_CY=${ntiles}
 
-CDATE="${PDY}${cyc}" ${CYCLESH}
-export err=$?; err_chk
+# Copy fix files required by global_cycle to DATA just once
+for (( nn=1; nn <= ntiles; nn++ )); do
+  ${NCP} "${FIXgfs}/orog/${CASE}/${CASE}_grid.tile${nn}.nc"                 "${DATA}/fngrid.00${nn}"
+  ${NCP} "${FIXgfs}/orog/${CASE}/${CASE}.mx${OCNRES}_oro_data.tile${nn}.nc" "${DATA}/fnorog.00${nn}"
+done
 
+# Copy the NSST analysis file for global_cycle
+# There is only a single NSST analysis at the middle of the window
+# For now use/assume it is the same at the beginning of the window if doing IAU
+if [[ "${DONST}" == "YES" ]]; then
+  ${NCP} "${COMIN_ATMOS_ANALYSIS}/${APREFIX}dtfanl.nc" "${DATA}/dtfanl"
+  export NST_FILE="dtfanl"
+else
+  export NST_FILE="NULL"
+fi
 
-################################################################################
-# Postprocessing
-cd ${pwd}
-[[ ${mkdata} = "YES" ]] && rm -rf ${DATA}
+# Collect the dates in the window to update surface restarts
+gcycle_dates=("${PDY}${cyc}")  # Always update surface restarts at middle of window
+if [[ "${DOIAU:-}" == "YES" ]]; then  # Update surface restarts at beginning of window
+  half_window=$(( assim_freq / 2 ))
+  BDATE=$(date --utc -d "${PDY} ${cyc} - ${half_window} hours" +%Y%m%d%H)
+  gcycle_dates+=("${BDATE}")
+fi
+
+# Loop over the dates in the window to update the surface restarts
+for gcycle_date in "${gcycle_dates[@]}"; do
+
+  echo "Updating surface restarts for ${gcycle_date} ..."
+
+  datestr="${gcycle_date:0:8}.${gcycle_date:8:2}0000"
+
+  # Copy inputs from COMIN to DATA
+  for (( nn=1; nn <= ntiles; nn++ )); do
+    ${NCP} "${sfcdata_dir}/${datestr}.sfc_data.tile${nn}.nc" "${DATA}/fnbgsi.00${nn}"
+    ${NCP} "${DATA}/fnbgsi.00${nn}"                       "${DATA}/fnbgso.00${nn}"
+  done
+
+  CDATE="${gcycle_date}" ${CYCLESH}
```

Comment:
@CoryMartin-NOAA 
My guess is the same as yours.
Perhaps reverting to:
```suggestion
  CDATE="${PDY}${cyc}" ${CYCLESH}
```
is necessary.

Reply:
@aerorahul I thought that but it works for the 00z cycle. I'm trying to dig in and see what exactly it is failing on.

Reply:
Snow obs are funny.  IIRC, they come in after the 18z cycle and are used in the 00z cycle.
Also, IIRC, the same obs are used for the rest of they cycles in the day.
I may be (and likely am) completely out of touch on this, so that's that.

Reply:
there's no DA in this particular test though, it's just running through the global_cycle